src/lib.rs

```diff
@@ -271,6 +271,62 @@ impl PolicyServer {
     }
 }
 
+// Load the ServerConfig to be used by the Policy Server configuring the server
+// certificate and mTLS when necessary
+//
+// RustlsConfig does not offer a function to load the client CA certificate together with the
+// service certificates. Therefore, we need to load everything and build the ServerConfig
+async fn build_tls_server_config(tls_config: &TlsConfig) -> Result<rustls::ServerConfig> {
+    use std::{fs::File, io::BufReader};
+
+    use rustls::{server::WebPkiClientVerifier, RootCertStore, ServerConfig};
+    use rustls_pemfile::Item;
+    use rustls_pki_types::{CertificateDer, PrivateKeyDer};
+
+    let cert_file = &mut BufReader::new(File::open(tls_config.cert_file.clone()).unwrap());
+    let key_file = &mut BufReader::new(File::open(tls_config.key_file.clone()).unwrap());
+    let cert = rustls_pemfile::certs(cert_file)
+        .map(|it| it.map(|it| it.to_vec()))
+        .collect::<Result<Vec<_>, _>>()?;
+    let mut key_vec: Vec<Vec<u8>> = rustls_pemfile::read_all(key_file)
+        .filter_map(|i| match i.ok()? {
+            Item::Sec1Key(key) => Some(key.secret_sec1_der().to_vec()),
+            Item::Pkcs1Key(key) => Some(key.secret_pkcs1_der().to_vec()),
+            Item::Pkcs8Key(key) => Some(key.secret_pkcs8_der().to_vec()),
+            _ => None,
+        })
+        .collect();
+    if key_vec.len() != 1 {
+        return Err(anyhow!("Cannot parse server key"));
+    }
+
+    let cert = cert.into_iter().map(CertificateDer::from).collect();
+    let key = PrivateKeyDer::try_from(key_vec.pop().unwrap())
+        .map_err(|e| anyhow!("Cannot parse server key: {e}"))?;
+
+    let config = if let Some(client_ca_cert_file_path) = tls_config.client_ca_cert_file.clone() {
+        // we have the client CA. Therefore, we should enable mTLS.
+        let client_ca_cert_file =
+            &mut BufReader::new(File::open(client_ca_cert_file_path).unwrap());
+
+        let mut ca_certs = RootCertStore::empty();
+        let client_ca_certs: Vec<_> = rustls_pemfile::certs(client_ca_cert_file)
+            .filter_map(|it| it.ok())
```

Comment:
Here we are silently skipping all the certificates that errored while being read.

I would do something like:

```rust
.filter_map(|it| {
  if let Err(e) = it {
     warn!(error=e, "cannot convert client CA file, ignoring")
  }
  it.ok()
})
```